src/lib/passwordGenerator.ts

```diff
@@ -0,0 +1,67 @@
+interface passwordProps {
+  includeUpper : boolean;
+  includeLower : boolean;
+  includeSymbol : boolean;
+  includeNumber : boolean;
+}
+
+export default class PasswordGenerator {
+  characters : string = "";
+  passwordLen : number = 12;
+
+  constructor(passwordProps : passwordProps, pwdLength ?: number) {
+    const { includeUpper, includeLower, includeSymbol, includeNumber } =
+      passwordProps;
+    if (pwdLength) this.passwordLength = pwdLength;
+    this.upperCase = includeUpper;
+    this.lowerCase = includeLower;
+    this.symbols = includeSymbol;
+    this.number = includeNumber;
+  }
+
+  set upperCase(isUpperCase : boolean) {
+    if (isUpperCase) {
+      this.characters += "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    }
+  }
+
+  set lowerCase(isLowerCase : boolean) {
+    if (isLowerCase) {
+      this.characters += "abcdefghijklmnopqrstuvwxyz";
+    }
+  }
+
+  set symbols(isSymbol : boolean) {
+    if (isSymbol) {
+      this.characters += "!@#$%^&*";
```

Comment:
I'm trying to decide if this is too many special characters or not enough. I think all of these are commonly-available in different languages, systems, etc.

Probably any set of characters is going to have some illegal values for certain sites (or not contain required characters from other sites). I end up having to manually adjust passwords generated by Keychain Access and BitWarden all the time.

Reply:
I grabbed these values from BW chrome extension. However in the extension you can deselect adding symbols. In the website version I don't remember seeing symbols so I could either remove it or when I initiate the class I could pass in false for isSymbol.